src/pages/smart-contracts/setting-up-local-environment.md

```diff
@@ -0,0 +1,332 @@
+
+When you develop your contract, you will often redeploy your `.clar` files. The fastest way is to create a deployment script.
+
+In the same contract project folder as before, install dependencies:
+
+```bash
+npm i @stacks/network @stacks/transactions --saveDev
+```
+
+Next, create the deployment script file:
+
+```bash
+mkdir scripts
+cd scripts
+touch deploy-contract.js
+```
+
+Finally, paste the following code:
+
+```js
+const readline = require('readline');
+const fs = require('fs');
+const { makeContractDeploy, broadcastTransaction } = require('@stacks/transactions');
+const { StacksTestnet } = require('@stacks/network');
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+// set to local mocknet
+const network = new StacksTestnet();
+network.coreApiUrl = 'http://localhost:3999';
+
+rl.question('Contract to deploy: /contracts/ ', async function (contract) {
+  const fileUrl = `../contracts/${contract}.clar`;
+  console.log(`Deploying contract from file: ${fileUrl}`);
+
+  const txOptions = {
+    contractName: `${contract}-${+new Date()}`,
+    codeBody: fs.readFileSync(fileUrl).toString(),
+    senderKey: 'cb3df38053d132895220b9ce471f6b676db5b9bf0b4adefb55f2118ece2478df01',
+    network,
+  };
+
+  const transaction = await makeContractDeploy(txOptions);
+
+  const txResult = await broadcastTransaction(transaction, network);
+
+  if (typeof txResult === 'string') {
+    console.log(`New transactions was broadcasted: http://localhost:3000/txid/0x${txResult}`);
+  } else {
+    console.log(txResult);
+  }
+
+  rl.close();
+});
+
+rl.on('close', function () {
+  process.exit(0);
+});
+```
+
+-> The script uses [default testnet keys](https://github.com/blockstack/stacks-blockchain-api/blob/e6e445b2cd1055f4bf25af7af646405783d5877e/src/api/routes/debug.ts#L36-L52)
+
+You can now simply deploy a contract by running the script and entering the file your want to deploy (without the `.clar` extension):
+
+```bash
+node deploy-contract.js
+# enter: "counter"
+
+-> Contract to deploy: /contracts/ counter
+-> Deploying contract from file: ../contracts/counter.clar
+-> New transactions was broadcasted: http://localhost:3000/txid/0x65dac9d90092422a682204f44c74c3617647d5659e077f0d6052c4097825df8f
+-> Contract name: counter-1602885440723
+```
+
+-> The docker image you spun up in the previous step needs to still be running
+
+You will notice that the name of the new contract has a unix timestamp appended to it. This will ensure that you can redeploy the same file. It is a simple versioning mechanism. You have to make sure to always use the most recent contract name.
+
+## Step 5: Set up Stacks Authenticator
+
+Users of your web app will need to authenticate using their Secret Key before they could sign a Stacks transaction. The authentication needs to point to your local mocknet.
+
+Let's spin up the authentication service:
+
+```bash
+cd ..
+git clone https://github.com/blockstack/ux.git
```

Comment:
@wileyj here's another thing we could discuss: maybe this could be another docker image that we pull into the compose file for this tutorial?

Reply:
For this I'll have to take a closer look, i'm not familiar with what's going on in this repo

Reply:
update: totally doable. i'll have to submit a PR to get it ready but we can adjust instructions later once that is ready as part of the full-docker setup